persistence-cassandra/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/cassandra/CassandraReadSideHandler.scala

```diff
@@ -60,7 +66,13 @@ private[cassandra] abstract class CassandraReadSideHandler[Event <: AggregateEve
             }
           )
 
-        invoke(handler, elem).flatMap(executeStatements)
+        for {
+          statements <- invoke(handler, elem)
+          _ <- executeStatements(statements)
+          // important: only commit offset once read view
+          // statements has completed successfully
+          _ <- executeStatements(offsetStatement(elem.offset) :: Nil)
```

Comment:
Split user statement write and offset write into two atomic operations to guarantee at least once if executing any of the user statements fails.

Reply:
Ah, out of Akka-habit I assumed that the Java side only defers to the Scala but that should indeed also be fixed.